src/cmd/ksh93/sh/path.c

```diff
@@ -901,15 +901,19 @@ void path_exec(Shell_t *shp, const char *arg0, char *argv[], struct argnod *loca
                 opath = arg0;
             }
             path_spawn(shp, opath, argv, envp, libpath, 0);
+            // Command exists but failed to exec
+            if (shp->path_err != ENOENT) {
+                // Exit with status 126
```

Comment:
I would replace this line with something like "// Assume that any other error (e.g., EACCES) means the file was found but could not be executed."